services/hooks/src/data.js

```diff
@@ -221,5 +221,42 @@ LastFire.prototype.definition = function() {
   });
 };
 
+LastFire.expires = async function(Hook, now, n = 100) {
+  const hookIds = new Set();
+  let count = 0;
+  await Hook.scan({},
+    {
+      handler: async hook => {
+        const { hookId } = await hook.definition();
+        hookIds.add(hookId);
+      },
+    },
+  );
+
+  for (hookId of hookIds) {
+    const hookData = [];
+    await this.scan({
+      taskCreateTime: Entity.op.lessThan(now),
+      hookId: Entity.op.equal(hookId)}, {
+      limit: 500,
+      handler: async item => {
+        const { hookGroupId, taskId, taskCreateTime } = await item.definition();
+        hookData.push({hookGroupId, taskId, taskCreateTime});
+      },
+    },
+    );
+    hookData.sort((a, b) =>
+      new Date(b.taskCreateTime) - new Date(a.taskCreateTime));
+
+    for(let i=n;i<hookData.length;i++) {
```

Comment:
This is more better written as `for (let hook of hookData) {..}`, as it avoids calling `hookData.length` on every iteration.